calyx-frontend/src/attribute.rs

```diff
@@ -76,6 +76,9 @@ pub enum BoolAttr {
     #[strum(serialize = "fast")]
     /// https://github.com/calyxir/calyx/issues/1828
     Fast,
+    #[strum(serialize = "protected")]
+    /// Do I preserve this cell (and its uses) during optimization?
```

Comment:
Could you make this a little bit more clear? Maybe "Indicated that the cell should not be removed or shared during optimization".